lib/truffle/truffle/cext.rb

```diff
@@ -1186,6 +1186,47 @@ def rb_undef(mod, name)
     end
   end
 
+  def rb_uv_to_utf8(buf, uv)
+    ptr = Truffle::FFI::Pointer.new(buf)
+
+    if uv <= 0x7f
+      ptr.put_char(0,   uv        & 0xff        )
+      1
+    elsif uv <= 0x7ff
+      ptr.put_char(0, ((uv >> 6 ) & 0xff) | 0xc0)
+      ptr.put_char(1, ( uv        & 0x3f) | 0x80)
+      2
+    elsif uv <= 0xffff
+      ptr.put_char(0, ((uv >> 12) & 0xff) | 0xe0)
+      ptr.put_char(1, ((uv >> 6 ) & 0x3f) | 0x80)
+      ptr.put_char(2, ( uv        & 0x3f) | 0x80)
+      3
+    elsif uv <= 0x1fffff
+      ptr.put_char(0, ((uv >> 18) & 0xff) | 0xf0)
+      ptr.put_char(1, ((uv >> 12) & 0x3f) | 0x80)
+      ptr.put_char(2, ((uv >> 6 ) & 0x3f) | 0x80)
+      ptr.put_char(3, ( uv        & 0x3f) | 0x80)
+      4
+    elsif uv <= 0x3ffffff
+      ptr.put_char(0, ((uv >> 24) & 0xff) | 0xf8)
+      ptr.put_char(1, ((uv >> 18) & 0x3f) | 0x80)
+      ptr.put_char(2, ((uv >> 12) & 0x3f) | 0x80)
+      ptr.put_char(3, ((uv >> 6 ) & 0x3f) | 0x80)
+      ptr.put_char(4, ( uv        & 0x3f) | 0x80)
+      5
+    elsif uv <= 0x7fffffff
+      ptr.put_char(0, ((uv >> 30) & 0xff) | 0xfc)
+      ptr.put_char(1, ((uv >> 24) & 0x3f) | 0x80)
+      ptr.put_char(2, ((uv >> 18) & 0x3f) | 0x80)
+      ptr.put_char(3, ((uv >> 12) & 0x3f) | 0x80)
+      ptr.put_char(4, ((uv >> 6 ) & 0x3f) | 0x80)
+      ptr.put_char(5, ( uv        & 0x3f) | 0x80)
+      6
+    else
+      raise RangeError, 'pack(U): value out of range'
+    end
+  end
```

Comment:
Interesting choice to write this in Ruby.
Using the original C definition might be slightly more efficient, but this seems fine too.